### PR TITLE
Added karma xkcd command

### LINKED ISSUE #198: xkcd style karma plots
petition for someone who knows wtf is going on with karma's code base to make this the output of !karma plot

### PR DESCRIPTION
**Pull request by henry50**
Adds `!karma xkcd` which is exactly the same as `!karma plot` but in the style of xkcd. The command definitions can probably be refactored more nicely, especially the duplicate error handlers, but I couldn't really see how.

Closes #198

### CODE REVIEW

**Comment by ericthelemur** [comment]
Thanks for the PR! I'll give it a proper review tonight, but looks good at a glance. To share the error handlers annotate it with both:

```
@plot.error
@xkcd.error
def command_error(...) ...
```

File: cogs/commands/karma.py [hunk 1]
```diff
@@ -473,13 +485,32 @@ async def plot(self, ctx: Context, *, args: str):
         file = File(img, filename=filename)
         await ctx.send(f"Here you go, {name}! {emoji}", embed=embed, file=file)
 
+    @karma.command(help="Plots the karma change over time of the given karma topic(s)")
+    @commands.cooldown(5, 60, BucketType.user)
+    async def plot(self, ctx: Context, *, args: str):
+        await self.common_plot(ctx, args, False)
+
     @plot.error
```
Comment (ericthelemur) [inline cogs/commands/karma.py:493]: Add xkcd error here too (might have to move after main xkcd)
```suggestion
    @plot.error
    @xkcd.error
```

File: cogs/commands/karma.py [hunk 2]
```diff
@@ -473,13 +485,32 @@ async def plot(self, ctx: Context, *, args: str):
...
     async def plot_error_handler(self, ctx: Context, error: KarmaError):
         if hasattr(error, "message"):
             await ctx.send(error.message)
         else:
             raise error
```
Comment (ericthelemur) [inline cogs/commands/karma.py:498]: This is not directly related, but when testing removing this raise made the error handling fallback to global correctly. Would be great if we could just roll that in
```suggestion
```

File: cogs/commands/karma.py [hunk 3]
```diff
@@ -473,13 +485,32 @@ async def plot(self, ctx: Context, *, args: str):
...
 
+    @karma.command(
+        help="Plots the karma change over time of the given karma topic(s) in the style of xkcd"
+    )
+    @commands.cooldown(5, 60, BucketType.user)
+    async def xkcd(self, ctx: Context, *, args: str):
+        await self.common_plot(ctx, args, True)
+
+    @xkcd.error
```
Comment (ericthelemur) [inline cogs/commands/karma.py:507]: Merge with plot error handler above
```suggestion
```

File: resources/Humor-Sans.ttf
Comment (ericthelemur) [inline resources/Humor-Sans.ttf:1]: Not sure what the licensing on directly including the font is, would be worth checking before directly adding it to source.

**Review by ericthelemur — COMMENTED** [review]
Almost all looks good to me

File: resources/Humor-Sans.ttf
Comment (henry50) [reply resources/Humor-Sans.ttf:1]: I couldn't find anything about licensing. The font could instead be installed in the Dockerfile by adding
```dockerfile
# install xkcd font
RUN apt-get update
RUN apt-get install fonts-humor-sans
```
and removing Humor-Sans.tff and
```python
font_path = str(Path("resources", "Humor-Sans.ttf"))
font_manager.fontManager.addfont(font_path)
```

Comment (ericthelemur) [reply resources/Humor-Sans.ttf:1]: I've had a bit of a look too since, and can't find info either. While I couldn't find an explicit license on it, I was mainly worried if it had a license that limited methods of distribution (something like ffmpeg).

I don't see any issue with including it since it was released out to the community and we are using it in an obviously intended way.

**Comment by ericthelemur** [comment]
Will give a final test and merge in the morning.

**Comment by ericthelemur** [comment]
This has been deployed, ty
